core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -2150,8 +2152,11 @@ private static class TimestampAddConvertlet implements SqlRexConvertlet {
       switch (call.operandCount()) {
       case 2:
         // Oracle-style 'ADD_MONTHS(date, integer months)'
-        if (call.getOperator() == SqlLibraryOperators.ADD_MONTHS) {
-          qualifier = new SqlIntervalQualifier(TimeUnit.MONTH, null, SqlParserPos.ZERO);
+        if (call.getOperator() == SqlLibraryOperators.ADD_MONTHS
```

Comment:
Use another condition to do this. Don't need to merge it.

Reply:
I optimized the code to make it more readable.